openedx/core/lib/xblock_utils/__init__.py

```diff
@@ -464,11 +464,38 @@ def xblock_local_resource_url(block, uri):
     xblock_class = getattr(block.__class__, 'unmixed_class', block.__class__)
     if settings.PIPELINE_ENABLED or not settings.REQUIRE_DEBUG:
         return staticfiles_storage.url('xblock/resources/{package_name}/{path}'.format(
-            package_name=xblock_class.__module__,
+            package_name=xblock_resource_pkg(xblock_class),
             path=uri
         ))
     else:
         return reverse('xblock_resource_url', kwargs={
             'block_type': block.scope_ids.block_type,
             'uri': uri,
         })
+
+
+def xblock_resource_pkg(block):
+    """
+    Return the module name need to find an XBlock's shared static assets.
+
+    This method will return the full module name that is one level higher than
+    the one the block is in. For instance, problem_builder.answer.AnswerBlock
+    has a __module__ value of 'problem_builder.answer'. This method will return
+    'problem_builder' instead. However, for edx-ora2's
+    openassessment.xblock.openassessmentblock.OpenAssessmentBlock, the value
+    returned is 'openassessment.xblock'.
+
+    XModules are special cased because they're local to this repo and they
+    actually don't share their resource files when compiled out as part of the
+    XBlock asset pipeline. This only covers XBlocks and XModules using the
```

Comment:
Question: how many XModules do we have today that still use xmodule_assets?  Does it simplify the code and upcoming changes to update those XModules?

Reply:
We have five XBlocks/XModules that use the XBlock-style asset pipeline but have their code in edx-platform -- LibraryContent, SplitTest, Vertical, WordCloud, Wrapper (?). Making them use the XModule-style super-bundle again would be a step backwards for overall end-user performance. At some point, they'll need to be migrated to something else, and I didn't think it was worth doing any moving around until then.